### PR TITLE
order confirmation popup

### PR DESCRIPTION
**Pull request by desafree**
create the order confirmation popup. Because the payment functionality is not implemented, after the checkout the user will be redirected to the homepage and the cart will be cleared

### CODE REVIEW

File: src/checkout/components/ItemsRecap/ItemsRecap.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { Grid } from "../../../_shared";
+import {
+  calculateTotalFromItems,
+  useSelectProductsFromCart,
+} from "../../../cart";
+import { OrderDetailRecap } from "../OrderDetailRecap";
+import { OrderTotalRecap } from "../OrderTotalRecap";
+
+export const ItemsRecap = () => {
+  const items = useSelectProductsFromCart();
```
Comment (rams23) [inline src/checkout/components/ItemsRecap/ItemsRecap.tsx]: useSelect discloses implementation details that are not necessary, i would rather use something like "useCartProducts" or "useProducts in cart", or just useCart, returning items as an object

File: src/checkout/components/OrderDetailRecap/OrderDetailRecap.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  items: ItemType[];
+}
+
+export const OrderDetailRecap: FC<Props> = ({ items }) => {
+  const { on, toggle } = useToggle();
+  const translate = useTranslate();
+
+  return (
+    <OrderDetailRecapStyled>
+      {!on ? (
```
Comment (rams23) [inline src/checkout/components/OrderDetailRecap/OrderDetailRecap.tsx]: renaming on as viewMore will make this component a lot clearer

File: src/checkout/components/OrderRecap/OrderRecap.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export const OrderRecap: FC<Props> = ({ toggle }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  useClickOutsideListener(ref, toggle);
+
+  return (
+    <OrderRecapStyled>
+      <OrderRecapContainer ref={ref} />
+    </OrderRecapStyled>
+  );
+};
```
Comment (rams23) [inline src/checkout/components/OrderRecap/OrderRecap.tsx]: can we use the popup component in this situation?

File: src/checkout/views/Checkout/Checkout.tsx [hunk 4]
```diff
@@ -6,42 +6,52 @@ import {
...
 } from "../../../_shared";
 import { CheckOutData } from "../../types";
 import { schemaCheckoutForm, defaultValuesForm } from "../../utils";
-import { FormOptions, Summary } from "../../components";
+import { FormOptions, Summary, OrderRecap } from "../../components";
+import { useToggle } from "../../../cart";
 
 export const Checkout = () => {
   const translate = useTranslate();
+  const { toggle, on } = useToggle();
```
Comment (rams23) [inline src/checkout/views/Checkout/Checkout.tsx]: also this "on" could be something like "showRecap"

**Review by rams23 — CHANGES_REQUESTED** [review]

File: src/checkout/components/OrderRecap/OrderRecap.tsx (hunk 3 above)
Comment (desafree) [reply src/checkout/components/OrderRecap/OrderRecap.tsx]: do you mean in terms of content or for the popup functionality itself?

**Comment by desafree** [comment]
Not so sure about using React.Children.map + React.cloneElement to pass ref to my popup container (passed as a child)

**Review by rams23 — APPROVED** [review]

**Pull request merged**